akka-stream-tests/src/test/scala/akka/stream/scaladsl/AcknowledgeSinkSpec.scala

```diff
@@ -21,18 +21,15 @@ class AcknowledgeSinkSpec extends AkkaSpec {
 
   val noMsgTimeout = 300.millis
 
-  def assertSuccess(value: Any, fb: Future[Option[Any]]): Unit =
-    Await.result(fb, 1.second) should be(Some(value))
-
   "An AcknowledgeSink" must {
 
     "send the elements as result of future" in assertAllStagesStopped {
-      val queue = Source(List(1, 2, 3)).runWith(Sink.queue(3))
-      assertSuccess(1, queue.pull())
-      assertSuccess(2, queue.pull())
-      assertSuccess(3, queue.pull())
-      queue.pull().pipeTo(testActor)
-      expectMsg(None)
+      val expected = List(Some(1), Some(2), Some(3), None)
+      val queue = Source(expected.flatten).runWith(Sink.queue(3))
+      expected foreach { v ⇒
+        queue.pull() pipeTo testActor
+        expectMsg(v)
+      }
```

Comment:
Ok, so now it is both fixed and the test more robust. sounds like winwin to
me!  😊

-- 
Cheers,
√
On 11 Dec 2015 10:29, "Roland Kuhn" <notifications@github.com> wrote:

> In
> akka-stream-tests/src/test/scala/akka/stream/scaladsl/AcknowledgeSinkSpec.scala
> <https://github.com/akka/akka/pull/19138#discussion_r47335174>:
>
> >    "An AcknowledgeSink" must {
> >
> >      "send the elements as result of future" in assertAllStagesStopped {
> > -      val queue = Source(List(1, 2, 3)).runWith(Sink.queue(3))
> > -      assertSuccess(1, queue.pull())
> > -      assertSuccess(2, queue.pull())
> > -      assertSuccess(3, queue.pull())
> > -      queue.pull().pipeTo(testActor)
> > -      expectMsg(None)
> > +      val expected = List(Some(1), Some(2), Some(3), None)
> > +      val queue = Source(expected.flatten).runWith(Sink.queue(3))
> > +      expected foreach { v ⇒
> > +        queue.pull() pipeTo testActor
> > +        expectMsg(v)
> > +      }
>
> Yes, but the timeout was not the issue: until #19097
> <https://github.com/akka/akka/pull/19097> is merged this will fail
> sporadically.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/akka/akka/pull/19138/files#r47335174>.
>